### PR TITLE
HPCC-18673 Remove index filepos special casing from helper interfaces

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [x] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Tested by running regression tests, and also compiler regression suite checking output.

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review.

File: rtl/eclrtl/rtldynfield.cpp [hunk 1]
```diff
@@ -1256,7 +1261,6 @@ class GeneralRecordTranslator : public CInterfaceOf<IDynamicTransform>
         // This code COULD move into rtlfield.cpp?
         switch(destType->getType())
         {
-        case type_filepos:
```
Comment (richardkchapman) [inline rtl/eclrtl/rtldynfield.cpp]: Should type_blob be here?

File: rtl/eclrtl/rtlfield.cpp [hunk 2]
```diff
@@ -514,100 +514,84 @@ bool RtlIntTypeInfo::canExtend(char &fillChar) const
...
-size32_t RtlFileposTypeInfo::process(const byte * self, const byte * selfrow, const RtlFieldInfo * field, IFieldProcessor & target) const
+size32_t RtlBlobTypeInfo::process(const byte * self, const byte * selfrow, const RtlFieldInfo * field, IFieldProcessor & target) const
 {
     assertex(callback);
-    if (isUnsigned())
-        target.processUInt(callback->getFilePosition(nullptr), field);
-    else
-        target.processInt(callback->getFilePosition(nullptr), field);
-    return length;
+    throwUnexpected();
```
Comment (richardkchapman) [inline rtl/eclrtl/rtlfield.cpp]: Unexpected? Or unimplemented?

File: rtl/eclrtl/rtlfield.cpp [hunk 3]
```diff
@@ -514,100 +514,84 @@ bool RtlIntTypeInfo::canExtend(char &fillChar) const
...
-void RtlFileposTypeInfo::setCallback(IThorIndexCallback *_callback)
+void RtlBlobTypeInfo::setCallback(IThorIndexCallback *_callback)
 {
     callback = _callback;
 }
 
-int RtlFileposTypeInfo::compare(const byte * left, const byte * right) const
+int RtlBlobTypeInfo::compare(const byte * left, const byte * right) const
 {
     throwUnexpected();  // Not needed and unimplementable
```
Comment (richardkchapman) [inline rtl/eclrtl/rtlfield.cpp]: Is needed, if postfiltering...

File: tools/dumpkey/dumpkey.cpp [hunk 4]
```diff
@@ -207,10 +203,11 @@ int main(int argc, const char **argv)
                             if (fieldNum == (unsigned) -1)
                                 throw MakeStringException(0, "Requested output field '%s' not found", fieldNames.item(idx));
                             const RtlFieldInfo *field = inrec.queryOriginalField(fieldNum);
-                            if (field->type->getType() == type_filepos)
+                            if (field->type->getType() == type_blob)
                             {
                                 // We can't just use the original source field in this case (as output record does not have a special filepos)
```
Comment (richardkchapman) [inline tools/dumpkey/dumpkey.cpp]: comment still refers to filepos

**Comment by richardkchapman** [comment]
@ghalliday A few minor questions/comments

File: rtl/eclrtl/rtlfield.cpp (hunk 2 above)
Comment (ghalliday) [reply rtl/eclrtl/rtlfield.cpp]: UNIMPLEMENTED - I will change.

File: rtl/eclrtl/rtlfield.cpp (hunk 3 above)
Comment (ghalliday) [reply rtl/eclrtl/rtlfield.cpp]: Yes, they should all be UNIMPLEMENTED - I left it as work to be done in the future.

Comment (richardkchapman) [reply rtl/eclrtl/rtlfield.cpp]: Comment is wrong then...

File: tools/dumpkey/dumpkey.cpp [hunk 5]
```diff
@@ -208,7 +208,7 @@ int main(int argc, const char **argv)
                             const RtlFieldInfo *field = inrec.queryOriginalField(fieldNum);
                             if (field->type->getType() == type_blob)
                             {
-                                // We can't just use the original source field in this case (as output record does not have a special filepos)
+                                // We can't just use the original source field in this case (as blobs are only support in the input)
```
Comment (richardkchapman) [inline tools/dumpkey/dumpkey.cpp]: grammar (support should say supported)

**Pull request merged**